tools/tdbg/commands.go

```diff
@@ -222,25 +223,16 @@ func AdminDeleteWorkflow(c *cli.Context) error {
 	ctx, cancel := newContext(c)
 	defer cancel()
 
-	resp, err := adminClient.DeleteWorkflowExecution(ctx, &adminservice.DeleteWorkflowExecutionRequest{
+	_, err = frontendClient.DeleteWorkflowExecution(ctx, &workflowservice.DeleteWorkflowExecutionRequest{
```

Comment:
the DeleteWorkflowExecution() api in frontend client is not force delete, so it won't work for corrupted workflows.

Reply:
for normal workflow deletion, plz use the delete workflow execution command in temporal cli.

Reply:
>the DeleteWorkflowExecution() api in frontend client is not force delete, so it won't work for corrupted workflows for troubleshooting purpose.

yes but delete API directly use database layer API

Reply:
Is the concern here frontend service should not access database layer API directly? If so, [this PR](https://github.com/temporalio/temporal/pull/4453) will forward that call to history service for processing.